cmd/govim/internal/types/types.go

```diff
@@ -222,5 +222,23 @@ type Diagnostic struct {
 	Range    Range
 	Text     string
 	Buf      int
-	Severity int
+	Severity Severity
 }
+
+// Severity is the govim internal representation of the LSP DiagnosticSeverites
+type Severity int
```

Comment:
Just to check: we're defining a new type here to deal with the fact that the protocol type is float?

Reply:
Yes, that floating severity in the protocol is.. yeah.